src/client/hud.cpp

```diff
@@ -327,13 +326,29 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 				video::SColor color(255, (e->number >> 16) & 0xFF,
 										 (e->number >> 8)  & 0xFF,
 										 (e->number >> 0)  & 0xFF);
-				core::rect<s32> size(0, 0, e->scale.X, text_height * e->scale.Y);
 				std::wstring text = unescape_translate(utf8_to_wide(e->text));
 				core::dimension2d<u32> textsize = textfont->getDimension(text.c_str());
+#ifdef __ANDROID__
+				// The text size on Android is not proportional with the actual scaling
+				irr::gui::IGUIFont *font_scaled = g_fontengine->getFont(
+						g_fontengine->getDefaultFontSize() - 3);
```

Comment:
Text scaling does not work. Please apply `e->size.X` in this calculation. Suggestion for the code above:
```C++
				irr::gui::IGUIFont *textfont = font;
				unsigned int font_size = g_fontengine->getDefaultFontSize();
#ifdef __ANDROID__
				font_size -= 3;
#endif
				if (e->size.X > 0)
					font_size *= e->size.X;

				if (font_size != g_fontengine->getDefaultFontSize()) {
					textfont = g_fontengine->getFont(e->size.X * font_size);
				}
```
Replaces `font_scaled`.

Reply:
Once again I look at the code and it works the way I intended.